src/HotChocolate/ApolloFederation/src/ApolloFederation/EntityTypeInterceptor.cs

```diff
@@ -30,14 +31,46 @@ internal class EntityTypeInterceptor : TypeInterceptor
                 definition);
         }
 
+        public override void OnTypesInitialized(
+            IReadOnlyCollection<ITypeDiscoveryContext> discoveryContexts)
+        {
+            if (!_entityTypes.Any())
+            {
+                throw EntityType_NoEntities();
+            }
+        }
+
         public override void OnBeforeCompleteType(
             ITypeCompletionContext completionContext,
             DefinitionBase definition,
-            IDictionary<string, object> contextData) =>
+            IDictionary<string, object> contextData)
+        {
             AddMemberTypesToTheEntityUnionType(
                 completionContext,
                 definition);
 
+            AddServiceTypeToQueryType(
+                completionContext,
+                definition);
+        }
+
+        private void AddServiceTypeToQueryType(
+            ITypeCompletionContext completionContext,
+            DefinitionBase definition)
+        {
+            if (completionContext.IsQueryType == true &&
+                definition is ObjectTypeDefinition objectTypeDefinition)
+            {
+                var fieldDescriptor = ObjectFieldDescriptor.New(
+                    completionContext.DescriptorContext,
+                    WellKnownFieldNames.Service);
+                fieldDescriptor
+                    .Type<NonNullType<ServiceType>>()
+                    .Resolver(ctx => new ServiceType());
```

Comment:
You do not have to create an instance here :) 
Resolver(default(object)) should be enough i think